### PR TITLE
Run CI on Python 3.9 and 3.13 (drop 3.8 from testing matrix)

### PR DESCRIPTION
**Pull request by krassowski**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/16850
Fixes https://github.com/jupyterlab/jupyterlab/issues/13946

## Code changes

- [x] Update the minimum supported version of `setuptools` to `41.1.0`

## User-facing changes

None

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by krassowski** [comment]
45 tests failed in `js-debugger` group when run on 3.13. This looks related to https://github.com/jupyterlab/jupyterlab/issues/13903 but it might be a new issue.

**Comment by krassowski** [comment]
test-minimum-versions fails with:

```
× Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [30 lines of output]
      Traceback (most recent call last):
        File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 112, in get_requires_for_build_wheel
          backend = _build_backend()
        File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 77, in _build_backend
          obj = import_module(mod_path)
        File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/importlib/__init__.py", line 127, in import_module
          return _bootstrap._gcd_import(name[level:], package, level)
        File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
        File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
        File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
        File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
        File "<frozen importlib._bootstrap_external>", line 850, in exec_module
        File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
        File "/tmp/pip-build-env-i71xy7tx/overlay/lib/python3.9/site-packages/setuptools/__init__.py", line 20, in <module>
          from setuptools.dist import Distribution, Feature
        File "/tmp/pip-build-env-i71xy7tx/overlay/lib/python3.9/site-packages/setuptools/dist.py", line 35, in <module>
          from setuptools.depends import Require
        File "/tmp/pip-build-env-i71xy7tx/overlay/lib/python3.9/site-packages/setuptools/depends.py", line 7, in <module>
          from .py33compat import Bytecode
        File "/tmp/pip-build-env-i71xy7tx/overlay/lib/python3.9/site-packages/setuptools/py33compat.py", line [55](https://github.com/jupyterlab/jupyterlab/actions/runs/11739714131/job/32704765500?pr=16852#step:4:56), in <module>
          unescape = getattr(html, 'unescape', html_parser.HTMLParser().unescape)
      AttributeError: 'HTMLParser' object has no attribute 'unescape'
      [end of output]
```

It looks like we cannot use 40.0.8 on Python 3.9 but we have to because otherwise we get:

```
      The conflict is caused by:
          The user requested setuptools>=40.8.0
          The user requested (constraint) setuptools==40.1.0
```

The question is where is the `The user requested setuptools>=40.8.0` coming from?

**Comment by krassowski** [comment]
It looks like lifting the constraint to even newer version of `setuptools` should solve it.

**Comment by jtpio** [comment]
For reference the `setuptools` dependency was added in https://github.com/jupyterlab/jupyterlab/pull/16423.

From https://github.com/jupyterlab/jupyterlab/issues/16404#issuecomment-2145957163:

> If we end up adding pin we should make it >=40.1.0 as this is when find_namespace_packages was added.

As of today, the latest version is 75.5.0. I guess it would be fine to increase the lower bound, not sure which version we still need to support though.

**Comment by krassowski** [comment]
Also, the `Linux Tests / Test Minimum Versions (pull_request)` job is now failing in all other PRs too due to:

```
Collecting async-lru>=1.0.0 (from jupyterlab==4.4.0a0)
  Using cached async_lru-1.0.0.tar.gz (4.9 kB)
  Installing build dependencies: started
  Running command pip subprocess to install build dependencies
  Using pip 24.3.1 from /opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/pip (python 3.8)
    Link requires a different Python (3.8.18 not in: '>=3.9'): https://files.pythonhosted.org/packages/21/df/7c6bb83dcb45b35dc35b310d752f254211cde0bcd2a35290ea6e2862b2a9/setuptools-75.4.0-py3-none-any.whl (from https://pypi.org/simple/setuptools/) (requires-python:>=3.9)
    Link requires a different Python (3.8.18 not in: '>=3.9'): https://files.pythonhosted.org/packages/e2/73/c1ccf3e057ef6331cc6861412905dc218203bde46dfe8262c1631aa7fb11/setuptools-75.4.0.tar.gz (from https://pypi.org/simple/setuptools/) (requires-python:>=3.9)
    Link requires a different Python (3.8.18 not in: '>=3.9'): https://files.pythonhosted.org/packages/fe/df/88ccbee85aefbca071db004fdc8f8d2507d55d5a9dc27ebb93c92edb1bd8/setuptools-75.5.0-py3-none-any.whl (from https://pypi.org/simple/setuptools/) (requires-python:>=3.9)
    Link requires a different Python (3.8.18 not in: '>=3.9'): https://files.pythonhosted.org/packages/c8/db/722a42ffdc226e950c4757b3da7b56ff5c090bb265dccd707f7b8a3c6fee/setuptools-75.5.0.tar.gz (from https://pypi.org/simple/setuptools/) (requires-python:>=3.9)
  ERROR: Cannot install setuptools>=40.8.0 because these package versions have conflicting dependencies.

  The conflict is caused by:
      The user requested setuptools>=40.8.0
      The user requested (constraint) setuptools==40.1.0

  To fix this you could try to:
  1. loosen the range of package versions you've specified
  2. remove package versions to allow pip to attempt to solve the dependency conflict
```

File: .github/workflows/linuxjs-tests.yml [hunk 1]
```diff
@@ -13,10 +13,7 @@ jobs:
     name: JS
     strategy:
       matrix:
-        # Fix for https://github.com/jupyterlab/jupyterlab/issues/13903
-        include:
-          - group: js-debugger
-            python-version: '3.11'
```
Comment (jtpio) [inline .github/workflows/linuxjs-tests.yml:19]: Looks like this should also fix https://github.com/jupyterlab/jupyterlab/issues/13946

**Review by jtpio — APPROVED** [review]
Thanks!

**Comment by jtpio** [comment]
I updated the top-level comment to mention the update of `setuptools` to it's more visible.
